src/SmartComponents/DriftPage/DriftPage.js

```diff
@@ -37,6 +37,12 @@ export class DriftPage extends Component {
         this.props.loadEntities();
     }
 
+    componentDidUpdate(prevProps) {
+        if (prevProps.location.search !== '/' && this.props.location.search === '') {
```

Comment:
When would the `prevProps.location.search` contain `/`? The query string always start with `?`.

Did you wanted to set the History when there is the transition from some querystring to a cleared one, basically getting the event when the "Comparison" is being clicked from the side nav?

If yes, you can try this condition:
```
prevProps.location.search !== '' && this.props.location.search === ''
```

Reply:
> Personally I find the nav clearing the search as a feature, rather as a bug. slightly_smiling_face But there might be some use case when this is not desired.

This is something that we want. We want to keep the comparison if you navigate away to Baselines to do some stuff and then come back.

> When would the `prevProps.location.search` would be `/`? The query string always start with `?`.

If you are on the Comparison nav and then click the Comparison nav again, `props.location.search` is `/`. This function works because I can catch this very specific scenario and revert.

Reply:
> This is something that we want. We want to keep the comparison if you navigate away to Baselines to do some stuff and then come back.

I see, make sense.

> If you are on the Comparison nav and then click the Comparison nav again, props.location.search is /. This function works because I can catch this very specific scenario and revert.

The first condition there is "not equal" `prevProps.location.search !== '/'` which would be always true, as `search` (query string) is either empty or starts with `?`. The condition is then redundant.

When clicked on the nav:
```
# prevProps.location
Object { 
  search: "?system_ids=c7ec8791-8534-41ce-a905-f7ed79c0f5db&filter[state]=different,incomplete_data&sort=-state,fact", 
  pathname: "/", 
  hash: "", 
  key: "i5ov76"
}

# this.props.location
Object { pathname: "/", search: "", hash: "", state: undefined, key: "z8ewez" }
```

Reply:
Good catch! All fixed now.

Reply:
Is it possible to keep it within the limits of the transition (from some query string to none, i.e. clicking on the nav bar) with the condition noted above https://github.com/RedHatInsights/drift-frontend/pull/474#discussion_r636819389? The `componentDidUpdate` is called on any param update. If there would be no query string and an other param is changed, then the `setHistory` will be called each time. I'm not sure if calling `setHistory` several times with the same attributes would cause any problems.